### PR TITLE
Cleanup watching prebuilds

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Ensure we don't re-watch a prebuild every time it changes status.

## How to test

https://ft-watch-pd847b7810d.preview.gitpod-dev.com/prebuilds

1. Make sure prebuilds still run and report status fine

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx [hunk 1]
```diff
@@ -52,29 +52,30 @@ interface Props {
 export const PrebuildDetailPage: FC = () => {
     const { prebuildId } = useParams<Props>();
 
-    const { data: prebuild, isLoading: isInfoLoading, error, refetch } = usePrebuildQuery(prebuildId);
+    const { data: initialPrebuild, isLoading: isInfoLoading, error, refetch } = usePrebuildQuery(prebuildId);
+    const [currentPrebuild, setCurrentPrebuild] = useState<Prebuild | undefined>();
+    const prebuild = currentPrebuild ?? initialPrebuild;
```
Comment (filiptronicek) [inline components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx:57]: This I think makes the page so much simpler. We used to use `initialPrebuild` somewhere and `currentPrebuild` elsewhere, but there was not a good reason why. Unifying the "watched" and "initially fetched" prebuild into `prebuild` should make everything here more readable.

**Comment by filiptronicek** [comment]
This doesn't work all the time and sometimes disposes of a stream in the middle of the prebuild's progress without re-initiating it. I need to investigate why

File: components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx [hunk 2]
```diff
@@ -114,21 +109,21 @@ export const PrebuildDetailPage: FC = () => {
         return () => {
             disposable.dispose();
         };
-    }, [prebuildId, currentPrebuild?.status?.phase?.name]);
```
Comment (filiptronicek) [inline components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx:117]: this dependency made us re-watch the status whenever the prebuild's phase changed

File: components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts [hunk 3]
```diff
@@ -23,7 +23,12 @@ type LogEventTypes = {
  * @param taskId ID of the task to watch.
  */
 export function usePrebuildLogsEmitter(prebuildId: string, taskId: string) {
-    const [emitter] = useState(new ReplayableEventEmitter<LogEventTypes>());
+    const emitter = useMemo(
+        () => new ReplayableEventEmitter<LogEventTypes>(),
+        // We would like to re-create the emitter when the prebuildId or taskId changes, so that logs of old tasks / prebuilds are not mixed with the new ones.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [prebuildId, taskId],
+    );
```
Comment (filiptronicek) [inline components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts:31]: This is also important, it fixes a bug where logs would be duplicated with copies from previous prebuild runs when re-running.

**Review by geropl — APPROVED** [review]
Tested and works nicely! :heavy_check_mark:

**Pull request merged**